compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2472,13 +2471,29 @@ static LogicalResult checkShapeRank(Operation *op, StringRef operandName,
 
 LogicalResult AttentionOp::verify() {
   Operation *op = getOperation();
-  unsigned numOperands = getNumOperands();
-  unsigned rankToCompareWith = 3;
-  if (numOperands == 6)
+
+  int numInputs = getNumDpsInputs();
+  int numOutputs = getNumDpsInits();
+
+  if (numInputs != 4) {
+    return op->emitOpError(
+        "expected 4 input operands: Query, Key, Value and Scale");
+  }
+
+  if (numOutputs != 1 && numOutputs != 3) {
+    return op->emitOpError(
+        "expected 1 or 3 output operands: Output, [Max and Sum]");
+  }
+
+  bool isTiled = numOutputs == 3;
+
+  int64_t rankToCompareWith;
+  if (isTiled) {
     rankToCompareWith = 2;
-  else if (numOperands != 4)
-    return op->emitOpError("expected operand count 4 or 6, but got")
-           << numOperands;
+  } else {
+    rankToCompareWith = 3;
+  }
+
```

Comment:
Now that the attention op takes `AnyType`, there needs to be a check here that `Q`, `K`, and `V` are actually shaped types (because the `get*Type` helpers are just casting).

Reply:
Done